### PR TITLE
Update typescript to v.4.4

### PR DESCRIPTION
**Pull request by azangru**
## Description
Upgrade typescript to version 4.4 This will help solve a type problem in PR https://github.com/Ensembl/ensembl-client/pull/631 correctly.

### Notable changes:
Typescript 4.4 switched from assigning type `any` to errors in the `catch` clause to assigning to them the type `unknown`. This required some code modifications on our part.

### CODE REVIEW

File: src/ensembl/src/content/app/browser/browserHelper.ts [hunk 1]
```diff
@@ -191,9 +192,14 @@ export const validateRegion = async (params: {
         onError
       );
     } catch (error) {
-      if (error.status === 400) {
+      const processibleError =
+        error &&
```
[the remark this answers is not in the document]
Comment (azangru) [reply src/ensembl/src/content/app/browser/browserHelper.ts:196]: Typescript isn't happy without it:

**Comment by azangru** [comment]
I regenerated package-lock.json 😢

File: src/ensembl/src/content/app/browser/browserHelper.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (msmanoj) [reply src/ensembl/src/content/app/browser/browserHelper.ts:196]: Ah! typeof null is `object` 😞

**Comment by azangru** [comment]
Another additions to this PR that surfaced after the regeneration of package-lock.json:
- We didn't have `redux-thunk` and `terser-webpack-plugin` in package.json, although we were referencing them in the code. I added those.
- There was a strange error `"Error: Cannot find module 'workbox-build/build/options/schema/webpack-generate-sw'"` related to webpack and workbox-webpack-plugin (see [link to failed job](https://gitlab.ebi.ac.uk/ensembl-web/ensembl-client/-/jobs/626589)). The suggested solution (see [Stack Overflow](https://stackoverflow.com/questions/68681211/vaadin-production-mode-builds-broken-in-v20#comment121377528_68681211)) is to update webpack-cli and workbox-webpack-plugin, so I did that. That fixed the error.

**Review by msmanoj — APPROVED** [review]

**Pull request merged**